code/modules/paperwork/paper_biscuit.dm

```diff
@@ -0,0 +1,125 @@
+			icon_state = "[icon_state]_cracked"
+			update_appearance()
+			return
+		else
+			return
+	if (cracked)
+		ui_interact(user)
+		return
+
+//Corporate "confidental" biscuit cards
+/obj/item/folder/biscuit/confidental
+	name = "\proper confidental biscuit card"
+	desc = "An confidental biscuit card. In a tasteful blue color with NT logo, looks like a chocolate bar. To reach contents you need to crack it open. Has label which says <b>DO NOT DIGEST</b>."
+	icon_state = "paperbiscuit_secret"
+	bg_color = "#355e9f"
+
+/obj/item/folder/biscuit/confidental/spare_id_safe_code
+	name = "\proper spare ID safe code biscuit card"
+	desc = "An biscuit card containing confidental spare ID safe code. In a tasteful blue color with NT logo, looks like a chocolate bar. To reach contents you need to crack it open. Has label which says <b>DO NOT DIGEST</b>."
+
+/obj/item/folder/biscuit/confidental/spare_id_safe_code/Initialize(mapload)
+	. = ..()
+	new /obj/item/paper/paperslip/corporate/fluff/spare_id_safe_code(src)
+
+/obj/item/folder/biscuit/confidental/emergency_spare_id_safe_code
+	name = "\proper spare emergency ID safe code biscuit card"
+	desc = "An biscuit card containing <i>not so confidental</i> emergency spare ID safe code. In a tasteful blue color with NT logo, looks like a chocolate bar. To reach contents you need to crack it open. Has label which says <b>DO NOT DIGEST</b>."
+
+/obj/item/folder/biscuit/confidental/emergency_spare_id_safe_code/Initialize(mapload)
+	. = ..()
+	new /obj/item/paper/paperslip/corporate/fluff/emergency_spare_id_safe_code(src)
+
+//Biscuits which start not-sealed/cracked initially for the crafting, printing and such
+/obj/item/folder/biscuit/not_sealed
+	name = "\proper biscuit card"
+	desc = "An biscuit card. To reach contents you need to crack it open. Has label which says <b>DO NOT DIGEST</b>."
+	icon_state = "paperbiscuit_cracked"
+	///What is the sprite for when its not cracked? As it starts already cracked, and for re-sealing needs to have a sprite
+	var/not_cracked_icon = "paperbiscuit"
+	cracked = TRUE
+	///Was the biscuit already sealed by players? To prevent several tgui alerts
+	var/sealed = FALSE
+
+/obj/item/folder/biscuit/not_sealed/examine()
+	. = ..()
+	if(!sealed)
+		. += span_notice("This one have never been sealed yet. Put in any contents to seal it by pressing it in hand. After sealing the only way to reach contents is by cracking it which is irreversible.")
+
+//Se
+/obj/item/folder/biscuit/not_sealed/attack_self(mob/user)
+	add_fingerprint(user)
+	if (!sealed)
+		if (tgui_alert(user, "Do you want to seal it? After sealing the only way to reach the contents is by cracking the biscuit, you cannot re-seal it again after that.", "Biscuit card", list("Yes", "No")) == "Yes")
+			cracked = FALSE
+			sealed = TRUE
+			playsound(get_turf(user), 'sound/items/duct_tape_snap.ogg', 60)
+			icon_state = "[not_cracked_icon]"
+			update_appearance()
+			return
+		else
+			return
```

Comment:
do the same thing i mentioned earlier with early returns and getting rid of the excess returns